### PR TITLE
Fix merging of plural keys

### PR DESCRIPTION
**Pull request by johnnyshields**

### CODE REVIEW

File: lib/i18n/js/utils.rb [hunk 1]
```diff
@@ -1,10 +1,16 @@
 module I18n
   module JS
     module Utils
+      PLURAL_KEYS = %i[zero one two few many other]
```
Comment (PikachuEXE) [inline lib/i18n/js/utils.rb]: Where does this come from?
Also it's not `frozen?`

File: lib/i18n/js/utils.rb [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+
       # deep_merge by Stefan Rusterholz, see <http://www.ruby-forum.com/topic/142809>.
       # The last result is modified to treat `nil` as missing key
       MERGER = proc do |_key, v1, v2|
-        Hash === v1 && Hash === v2 ? v1.merge(v2, &MERGER) : (v2.nil? ? v1 : v2)
+        if Hash === v1 && Hash === v2
+          (v2.keys - PLURAL_KEYS).empty? ? v2 : v1.merge(v2, &MERGER)
```
Comment (PikachuEXE) [inline lib/i18n/js/utils.rb]: Please add some code comment to explain the code here
Also please add test case

**Comment by johnnyshields** [comment]
This fixes the case that the key being merged is a pluralization. Suppose I have:

```
en:
  apple:
    one: Apple
    other: Apples

fr:
  apple:
    zero: Pomme
    one: Pomme
    other: Pommes
```

With fallbacks, this will merge to:

```
en:
  apple:
    zero: Pomme
    one: Apple
    other: Apples
```

**Comment by PikachuEXE** [comment]
Test case please 0w0

**Comment by PikachuEXE** [comment]
Let me know when test cases are added

**Comment by johnnyshields** [comment]
@PikachuEXE this is ready to merge

**Comment by johnnyshields** [comment]
@PikachuEXE this is ready to merge

**Comment by ghiculescu** [comment]
@johnnyshields this broke things in our app. Can you please confirm if https://github.com/fnando/i18n-js/pull/551 works for you.

**Comment by johnnyshields** [comment]
#551 does not work for me; see comment there. Recommend you to fix your app.

**Pull request merged**